src/data/CCImmutArray.mli

```diff
@@ -36,6 +36,11 @@ val get : 'a t -> int -> 'a
 val set : 'a t -> int -> 'a -> 'a t
 (** Copy the array and modify its copy *)
 
+val sub : 'a t -> int -> int -> 'a t
+(** [sub a start len] returns a fresh array of length len, containing the elements number start to start + len - 1 of array a.
+    Raises [Invalid_argument "Array.sub"] if start and len do not designate a valid subarray of a; that is, if start < 0, or len < 0, or start + len > Array.length a. *)
```

Comment:
- the comment has too long lines (80~100 chars max is more readable)
- please add `@since NEXT_RELEASE` to the end of the comment :-)